src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  ChangeEvent,
+  ComponentProps,
+  FocusEvent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useState,
+} from "react"
+import styles from "./SearchInput.module.css"
+
+interface SearchInputProps extends ComponentProps<"input"> {
+  searchIcon?: ReactNode
+  clearIcon?: ReactNode
+  onDebounce?: (searchTerm: string) => void
+  debounceDelay?: number
+  onSearchClick?: (searchTerm: string) => void
+  value?: string
+}
+
+export function SearchInput({
+  name,
+  placeholder,
+  searchIcon,
+  clearIcon,
+  onDebounce,
+  debounceDelay = 300,
+  onSearchClick,
+  value = "",
+  ...inputProps
+}: SearchInputProps) {
+  const [searchTerm, setSearchTerm] = useState(value)
```

Comment:
Saving props to state usually opens a can of worms when the props (`value`) change on the fly

Reply:
Yeah, I originally had it set via a useEffect. I think I was aiming for less code, but I totally see what you mean given the value is highly dynamic. I also really went back and forth on whether this component should control any kind of state. I could just as easily include an onClear callback and ensure all state management is contained in the parent.

Reply:
I think if the intention is to copy/paste this component as a sort of boilerplate/head start, it should be controlled (no local state). An implementor is going to have to use the `onChange` handler to get the value anyway (though I prefer @luiz-chagas suggestion of an `onSearch` handler and abstracting the input away entirely). If I'm passing `value` in, I don't expect anything else to manage it.

Reply:
Originally the intention was to have it be uncontrolled. You would use the `onDebounce` or @luiz-chagas suggested better name of `onSearch` and that was that. I then back peddled some and overcomplicated it by trying to marry some form of controlled/uncontrolled but it clearly became more fragile. If we want it to be fully controlled, I think we have an onChange, value, and onClear that are required. Then it is left to the user to leverage them in a way that makes sense.

Reply:
I'm leaning that way, so at that point this is just really a glorified input that sort of represents a search bar....which if I'm being honest, I probably wouldn't reach for. That said, I also don't want to create something that is fragile and overcomplicated.

Reply:
If we want to further strip it down, instead of a left and right icon with search related names, it could just be a left/right element. At that point you are responsible for any sort of events you would want to assign to those elements. 

Accept now it's just an input with a start adornment and end adornment. At that point it might as well be part of a larger composite like our existing TextInput, which is more like a TextField as it has helper text and a label. However, we are building everything in isolation so everything just gets recreated which just feels wrong. 

That said these thoughts are really just part of a larger conversation and reflect my strong desire to build something representing more of a library than a copy/paste repo.

Reply:
Thinking through this more, the case for a controlled, debounced component is more complicated when trying to keep it in sync with outside changes. I had to write this scenario down and think through the implementation to make much sense of either approach:

I have a search page at `/search` that I want to use a Search Input component on. Since I want to support deep-linking and browser history as people use the search page, the search query that I want send to my API is represented in the URL as `search?q=search%20term`.

## Expectations
* search results and search query are updated when the URL changes (from history event)
* URL is updated when `onSearch` is called (after debounced user input)

### Uncontrolled
* container passes value from `?q=` to the `value` prop
* container performs a route update on `onSearch`
* container does not pass `onChange` callback to Search Input
* Search Input needs to watch its own `value` prop with `useEffect` ([as you mentioned](https://github.com/cremalab/crema-components/pull/41/files#r946310242)) to `setSearchTerm` and call the debounce handler in order to keep the search term in sync with the URL param

### Controlled
* container manages search input's _actual current value_ in state somewhere, passes it to the `value` prop, updates it when Search Input's `onChange` event fires
* container performs a route update on `onSearch`
* container watches value from `?q=` and updates value sent to Search Input's `value` when it changes

I'm not clear on a pattern that would simplify this other than making it _feel_ controlled but using the "uncontrolled" approach above. 

🤷

Reply:
Had some thoughts over lunch. Let me play around. I'll ping you once I've implemented them :)